dnf-behave-tests/dnf/steps/lib/dnf.py

```diff
@@ -96,6 +96,12 @@ def find_transaction_table_begin(lines):
 
         return i + 2
 
+    # DNF 5 requires a different match
+    trans_start_re = re.compile(r"Package +Arch +Version +Repository +Size")
+    for i in range(0, len(lines) - 1):
+        if trans_start_re.match(lines[i]):
+            return i + 1
+
```

Comment:
Could you fork the code based on the `context.dnf.dnf5_mode` variable so that it's clearly separated on what is used for dnf 4 and what is used for dnf 5?

Reply:
Sure. Changed.